lib/tasks/users.rake

```diff
@@ -4,4 +4,15 @@ namespace :users do
     exit if ENV['READ_ONLY'].present?
     User.where(currently_syncing: false).optin.order('last_synced_at ASC').where('last_synced_at < ?', 1.week.ago).limit(100).each(&:update_repo_permissions_async)
   end
+
+  task fix_auth_tokens: :environment do
+    exit if ENV['READ_ONLY'].present?
+    AuthToken.authorized.find_each do |token|
+      if token.still_authorized?
```

Comment:
I think my only concern, besides not really knowing what other code would be affected by this, is still_authorized? on AuthToken could return false if the token has hit its rate limit at the time this method is run and would be deleted when running the rake task. I don't know if that is really a valid concern since that is a very low chance of happening.

Reply:
rate limit doesn't return true or false it returns a struct. `#<struct Octokit::RateLimit limit=5000, remaining=4966, resets_at=2019-02-01 11:05:19 -0500, resets_in=1720>`. The remaining would just be 0. The check just makes sure you don't get an auth failure. Valid tokens still get added on user permission syncs from the accounts that we still have access to.